pipelines/_ingredients/ApPipeCalibrateImage.yaml

```diff
@@ -66,43 +71,63 @@ subsets:
+              detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).difference.name
+    msg: "subtractImages.difference != detectAndMeasure.difference"
+  - contract: subtractImages.connections.ConnectionsClass(config=subtractImages).science.name ==
+              detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).science.name
+    msg: "subtractImages.science != detectAndMeasure.science"
+  - contract: subtractImages.connections.ConnectionsClass(config=subtractImages).template.name ==
+              diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
+    msg: "subtractImages.template != diaPipe.template"
+  - contract: subtractImages.connections.ConnectionsClass(config=subtractImages).science.name ==
+              diaPipe.connections.ConnectionsClass(config=diaPipe).exposure.name
+    msg: "subtractImages.science != diaPipe.exposure"
+  - contract: detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).diaSources.name ==
+              filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).diaSourceCat.name
+    msg: "detectAndMeasure.diaSources != filterDiaSrcCat.diaSourceCat"
+  - contract: detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).subtractedMeasuredExposure.name ==
+              rbClassify.connections.ConnectionsClass(config=rbClassify).difference.name
+    msg: "detectAndMeasure.subtractedMeasuredExposure != rbClassify.difference"
+  - contract: detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).subtractedMeasuredExposure.name ==
+              transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diffIm.name
+    msg: "detectAndMeasure.subtractedMeasuredExposure != transformDiaSrcCat.diffIm"
+  - contract: detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).subtractedMeasuredExposure.name ==
+              diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
+    msg: "detectAndMeasure.subtractedMeasuredExposure != diaPipe.diffIm"
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              rbClassify.connections.ConnectionsClass(config=rbClassify).diaSources.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != rbClassify.diaSources"
+  - contract: filterDiaSrcCat.connections.ConnectionsClass(config=filterDiaSrcCat).filteredDiaSourceCat.name ==
+              transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
+    msg: "filterDiaSrcCat.filteredDiaSourceCat != transformDiaSrcCat.diaSourceCat"
+  - contract: (not transformDiaSrcCat.doIncludeReliability) or
+              (rbClassify.connections.ConnectionsClass(config=rbClassify).classifications.name ==
+               transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).reliability.name)
+    msg: "rbClassify.classifications != transformDiaSrcCat.reliability"
```

Comment:
Might want to change the message a bit to allow for the first case (not transformDiaSrcCat.doIncludeReliability) to be the trigger as opposed to only the second case. This comment also goes for the other instances of it.

Reply:
But if `doIncludeReliability` isn't set, then `transformDiaSrcCat.reliability` doesn't even exist (and wouldn't be used if it did), so the contract is trivially satisfied.